connection.go

```diff
@@ -249,6 +273,20 @@ func (sc *snowflakeConn) Close() (err error) {
 	sc.cleanup()
 	return nil
 }
+
+func WithContextQueryTag(ctx context.Context, qt *QueryTag) context.Context {
```

Comment:
can we make it so that users can set other parameters too? so maybe `WithContextStatementParameters` and `GetContextStatementParameters`, where the value is a `map[string]string` - I really don't think the `gosnowflake` library should be responsible for serializing our requestId+email json blob

Reply:
+1 to @donhcd's comment here. Making this more general seems like a good addition.

Reply:
I've updated it to use a map instead. If new params are needed, a new context will be created that merges the inputs.